util/include/Metrics.hpp

```diff
@@ -65,17 +73,41 @@ class Aggregator {
 
 // A Gauge is a an integer value that shows the current value of something. It
 // can only be varied by directly setting and getting it.
-class Gauge {
+template <class T>
+class basic_gauge {
  public:
-  explicit Gauge(const uint64_t val) : val_(val) {}
-
+  explicit basic_gauge(const uint64_t val) : val_(val) {}
+  basic_gauge(const basic_gauge& gauge) { val_ = (unsigned long)gauge.val_; }
```

Comment:
I believe what you are doing here is undefined behavior. Presumably the reason you needed the cast was because atomics have deleted copy constructors. This should tell you that the type is not copyable on purpose. 

Why do you need to be able to copy metrics types in the first place?

In case you actually did need to copy them for some reason, you can use [constexpr if](https://en.cppreference.com/w/cpp/language/if#Constexpr_If) and then call `gauge.val_.load()` rather than casting. 

My recommendation is to remove the copy constructor/assignment ops altogether, and try to break out of the habit of using C-style casts :)

Reply:
atomic_uint64_t overloads the ‘unsigned long’ operator, sonI believe the code is correct.
I need a copy constructor because we do copy the values every time we call to updateAggregator

Reply:
I guess I'm a bit confused. Yes, there is an operator, but does this code call it? I believe the C-style cast just returns the memory pointed to by `gauge.val_` without running any code.

Reply:
Related. Unsigned long doesn't necessarily map to uint64_t.

Reply:
Ah, it appears the cast does call the operator. My bad. :)  Can we cast it as a `uint64_t` though to ensure it works on all architectures?

Reply:
OK, I will